RUFAS/biophysical/manure/storage/compost_bedded_pack_barn.py

```diff
@@ -0,0 +1,307 @@
+
+        Parameters
+        ----------
+        storage_nitrous_oxide_N : float
+            The nitrogen loss through nitrous oxide emissions on the current day, kg.
+        storage_ammonia_N : float
+            The nitrogen loss through ammonia emissions on the current day, kg.
+        storage_N_loss_from_leaching : float
+            The nitrogen loss through leaching on the current day, kg.
+
+        Raises
+        ------
+        ValueError
+            If the total nitrogen losses are greater than the total received manure nitrogen.
+        """
+        received_manure_nitrogen_after_losses = (
+            self._manure_to_process.nitrogen
+            - storage_nitrous_oxide_N
+            - storage_ammonia_N
+            - storage_N_loss_from_leaching
+        )
+        if received_manure_nitrogen_after_losses < 0:
+            self._om.add_error(
+                "Nitrogen loss application error",
+                "Cannot have total nitrogen losses greater than total received manure nitrogen.",
+                info_map={"class": self.__class__.__name__, "function": self._apply_nitrogen_losses.__name__},
+            )
+            raise ValueError(
+                "Nitrogen loss application error: cannot have total nitrogen losses greater than "
+                "total received manure nitrogen."
+            )
+        self._manure_to_process.ammoniacal_nitrogen = max(
+            0.0, self._manure_to_process.ammoniacal_nitrogen - storage_ammonia_N
+        )
+        self._manure_to_process.nitrogen = received_manure_nitrogen_after_losses
+
+    @staticmethod
+    def _calculate_cbpb_nitrous_oxide_emission(received_nitrogen: float, is_bedding_tilled: bool) -> float:
+        """
+        Calculate the nitrogen loss from nitrous oxide emission in a compost bedded pack barn.
+
+        Parameters
+        ----------
+        received_nitrogen : float
+            The mass of nitrogen present in the manure excreted by animals (kg).
+        is_bedding_tilled : bool
+            Indicator for if the bedding is tilled for the current simulation day.
+
+        Returns
+        -------
+        float
+            The nitrogen lost to nitrous oxide emissions in the compost bedded pack barn (kg).
+
+        Raises
+        ------
+        ValueError
+            If the daily nitrogen input is negative.
+
+        """
+
+        if received_nitrogen < 0.0:
+            raise ValueError(f"Daily nitrogen input mass must be non-negative: {received_nitrogen}")
+
+        coefficient_tilled = NITROUS_OXIDE_COEFFICIENT_WITH_TILLED_BEDDING
+        coefficient_untilled = NITROUS_OXIDE_COEFFICIENT_WITH_UNTILLED_BEDDING
+
+        nitrogen_loss_tilled = coefficient_tilled * received_nitrogen * is_bedding_tilled
+        nitrogen_loss_untilled = coefficient_untilled * received_nitrogen * (not is_bedding_tilled)
+
+        return nitrogen_loss_tilled + nitrogen_loss_untilled
```

Comment:
```suggestion
        coefficient = NITROUS_OXIDE_COEFFICIENT_WITH_TILLED_BEDDING if is_bedding_tilled else NITROUS_OXIDE_COEFFICIENT_WITH_UNTILLED_BEDDING

        return coefficient * received_nitrogen
```
I think this would be easier to understand, and much cleaner.